src/mock/quick_p2p/network.rs

```diff
@@ -218,13 +216,27 @@ impl Queue {
         self.0.push_back(packet)
     }
 
-    fn pop(&mut self) -> Option<Packet> {
-        self.0.pop_front()
-    }
-
     fn is_empty(&self) -> bool {
         self.0.is_empty()
     }
+
+    // This function will pop random msg from the queue.
+    fn pop(&mut self, rng: &mut SeededRng) -> Option<Packet> {
+        let mut first_non_random_msg = 0;
+        for packet in self.0.iter() {
```

Comment:
Maybe this for loop can be replaced with
```
let first_non_random_msg = self.0.iter().position(|packet| packet == Packet::Disconnect).unwrap_or(0);
```
(or some variation if `Packet` isn't `PartialEq`)